src/components/encoding-pane/filter-pane.tsx

```diff
@@ -0,0 +1,226 @@
+  schema: Schema;
+  handleAction?: (action: FilterAction) => void;
+}
+
+interface FilterPaneProps extends FilterPaneDropTargetProps, FilterPanePropsBase {};
+
+class FilterPaneBase extends React.Component<FilterPaneProps, {}> {
+  // private index: number;
+
+  public constructor(props: FilterPaneProps) {
+    super(props);
+    this.filterModifyTimeUnit = this.filterModifyTimeUnit.bind(this);
+  }
+
+  public render() {
+    const {filters, connectDropTarget} = this.props;
+    const filterShelves = filters.map((filter, index) => {
+      return this.renderFilterShelf(filter, index);
+    });
+    return connectDropTarget(
+      <div>
+        {filterShelves}
+        {this.fieldPlaceholder()}
+      </div>
+    );
+  }
+
+  protected filterRemove(index: number) {
+    const {handleAction} = this.props;
+    handleAction({
+      type: FILTER_REMOVE,
+      payload: {
+        index: index
+      }
+    });
+  }
+
+  protected filterModifyTimeUnit(timeUnit: string, index: number) {
+    const {handleAction} = this.props;
+    if (timeUnit === '-') {
+      timeUnit = undefined;
+    }
+
+    handleAction({
+      type: FILTER_MODIFY_TIME_UNIT,
+      payload: {
+        index: index,
+        timeUnit: timeUnit as TimeUnit
+      }
+    });
+  }
+
+  private renderFilterShelf(filter: RangeFilter | OneOfFilter, index: number) {
+    const {fieldDefs, schema, handleAction} = this.props;
+    const fieldIndex = schema.fieldNames().indexOf(filter.field);
+    const fieldDef = fieldDefs[fieldIndex];
+    let domain = schema.domain(fieldDef as FieldQuery);
+    const popupComponent =
+      fieldDef.type === ExpandedType.TEMPORAL &&
+      <FunctionPicker
+        fieldDef={fieldDef}
+        filter={filter}
+        index={index}
+        onFunctionChange={this.filterModifyTimeUnit}
+      />;
+    let filterComponent;
+    if (fieldDef.type === ExpandedType.QUANTITATIVE) {
+      // quantitative range filter
+      filterComponent = (
+        <RangeFilterShelf
+          domain={domain}
+          index={index}
+          filter={filter as RangeFilter}
+          type={fieldDef.type}
+          handleAction={handleAction}
+        />
+      );
+    } else if (fieldDef.type === ExpandedType.TEMPORAL) {
+      const timeUnit = filter.timeUnit;
+      if (timeUnit) {
+        if (timeUnit === TimeUnit.MONTH || timeUnit === TimeUnit.DAY) {
+          // time unit one of filter
+          domain = getDefaultList(timeUnit);
+          filterComponent = (
+            <OneOfFilterShelf
+              domain={domain}
+              index={index}
+              filter={filter as OneOfFilter}
+              handleAction={handleAction}
+            />
+          );
+        } else {
+          // time unit range filter
+          domain = getDefaultRange(domain, timeUnit);
+          let type;
+          if (timeUnit === TimeUnit.YEARMONTHDATE) {
+            type = ExpandedType.TEMPORAL;
+          } else {
+            type = ExpandedType.QUANTITATIVE;
```

Comment:
I set the type to TEMPORAL so that it'd render date time pickers. Set it to quantitative just renders inputs.